server/bin/artemis/configure-artemis.py

```diff
@@ -0,0 +1,311 @@
+    if os.path.exists(artemis_install_path):
+        logger.debug("Removing artemis installation: %s" % (artemis_install_path))
+        shutil.rmtree(artemis_install_path)
+
+def uninstall_service():
+    call("sudo semodule -vr artemisservice", "", allow_failure=True)
+    # to_delete = ["%s/artemisservice.mod" % BASE_DIR,
+    #              "%s/artemisservice.pp" % BASE_DIR,
+    #              "/usr/lib/systemd/system/artemis.service"]
+    #
+    # for next_file in to_delete:
+    #     if os.path.exists(next_file): os.remove(next_file)
+
+    if os.path.exists(SERVICE_FILE_PATH):
+        call("sudo systemctl disable artemis", "Failed to disable artemis service", allow_failure=True)
+        os.remove(SERVICE_FILE_PATH)
+    call("sudo systemctl daemon-reload", "Failed to reload systemd daemons.")
+    call("sudo systemctl reset-failed", "Failed to reset failed services.")
+
+    if os.path.exists(SELINUX_BUILD_DIR):
+        shutil.rmtree(SELINUX_BUILD_DIR)
+
+def parse_options():
+    usage = "usage: %prog"
+    parser = OptionParser(usage=usage)
+    parser.add_option("--version", action="store", type="string", default=DEFAULT_VERSION,
+                      help="the version of artemis to install")
+    parser.add_option("--broker-config", action="store", type="string",
+                      default="%s/../../src/main/resources/broker.xml" % (BASE_DIR),
+                      help="the broker config file to use.")
+    parser.add_option("--broker-name", action="store", type="string",
+                      default="candlepin",
+                      help="the name of the broker to be installed")
+    parser.add_option("--broker-data-dir", action="store", type="string",
+                      help="the location to store the broker's data files (defaults to broker's data dir)")
+    parser.add_option("--clean", action="store_true", help="clean current installation")
+    parser.add_option("--start", action="store_true", help="start broker after install")
+    parser.add_option("--debug", action="store_true", help="enables debug logging")
+
+    return parser.parse_args()
+
+def main():
+    (options, args) = parse_options()
+
+    if options.debug:
+        logger.setLevel("DEBUG")
+
+    if options.clean:
+        cleanup(options.version, INSTALL_DIR, BROKER_ROOT)
+        return
+
+    logger.info("Installing Artemis...")
+    artemis_path = install_artemis(options.version, INSTALL_DIR)
+    broker_path = create_broker(artemis_path, BROKER_ROOT, options.broker_name)
+    install_artemis_service(options.broker_name)
+    update_broker_config(broker_path, options.broker_config, options.broker_data_dir)
+    logger.info("Artemis was successfully installed!")
+
+    if options.start:
+        logger.info("Starting the Artemis server.")
+        call("sudo systemctl restart artemis", "Unable to start the artemis service.")
+
+if __name__ == "__main__":
+    main()
```

Comment:
Food for thought: we're probably going to move this file to Ansible given time, so that it's actions can be done ahead-of-time for Vagrant images and docker images. We can start with just calling this script, though.

Reply:
Another thing to think about. There's some one-time actions in this script (ex. install, start Artemis), and some multiple use stuff (cleaning, restarting Artemis)... It might be worth splitting out just a bit. Or not... Thoughts?